### PR TITLE
Document the release process

### PR DESCRIPTION
**Pull request by arkid15r**
## Proposed change

Document the release process

## Type of change

- [ ] New country/market holidays support (thank you!)
- [ ] Supported country/market holidays update (calendar discrepancy fix, localization)
- [x] Existing code/documentation/test/process quality improvement (best practice, cleanup, refactoring, optimization)
- [ ] Dependency update (version deprecation/upgrade)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Breaking change (a code change causing existing functionality to break)
- [ ] New feature (new python-holidays functionality in general)

## Checklist

- [x] I've followed the [contributing guidelines][contributing-guidelines]
- [x] I've run `make pre-commit`, it didn't generate any changes
- [x] I've run `make test`, all tests passed locally

[contributing-guidelines]: https://github.com/vacanza/python-holidays/blob/beta/CONTRIBUTING.rst
[docs]: https://github.com/vacanza/python-holidays/tree/beta/docs/source

### CODE REVIEW

**Review by KJhellico — DISMISSED** [review]
Excellent! LGTM 👍

File: holidays/RELEASE.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  - push changes to https://github.com/vacanza/python-holidays ``beta`` branch
+  - make sure the push related CI/CD job(s) have been completed successfully
+
+- Merge the finalized changes into ``master`` branch:
+
+  - create a new PR for the recent changes from ``beta`` to ``master`` branch
+    using 'v<version>' as a PR title and the previously generated release notes
+    as a PR description
+  - get the PR reviewed by at least one of the code owners
+  - merge the PR into ``master`` branch by enabling auto-merge with
```
Comment (KJhellico) [inline holidays/RELEASE.rst:24]: A bit unclear about "enabling auto-merge" at this stage.

File: holidays/RELEASE.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  - click on 'Choose a tag', enter 'v<version>' into the input field
+    (you should see something like 'Create a new tag: v0.39' on publish'
+  - select **master** - instead of default ``beta`` in 'Target' dropdown
+  - put 'v<version>' into 'Release title' field, e.g. 'v0.39'
+  - click on 'Generate release notes' button to collect new contributors and
+    full changelog link information (we normally keep it at the bottom with
+    a bit of re-formatting)
+  - replace auto-generated release notes with the previously generated release
+    notes (keep the new contributors and full changelog link)
+  - check/uncheck the 'Set as the latest release' checkbox depending on the
```
[the remark this answers is not in the document]
Comment (arkid15r) [reply holidays/RELEASE.rst:44]: This is for future (this year already!) situations when we'll have stable (say, v1.x) and legacy (v0.x) versions of PH

File: holidays/RELEASE.rst (hunk 1 above)
Comment (arkid15r) [reply holidays/RELEASE.rst]: 👍 
Removed that.

**Review by KJhellico — APPROVED** [review]

**Pull request merged**